### PR TITLE
ENH: Add MultiIndex.dtypes

### LINKED ISSUE #37062: ENH: MultiIndex.dtypes
create a mi
```
In [13]: mi = pd.MultiIndex.from_product([[1,2,3], list('abc'), pd.date_range('20200101', periods=2, tz='UTC')], names=['int', 'string', 'dt'])                            
```

You can already get .dtypes, but is slightly cumbersome. I would propose adding `Multidex.dtypes` (we *already* have MultiIndex.dtype but its *always* object). I think this is worth the convenience api.
```
In [14]: mi.to_frame().dtypes                                                                                                                                              
Out[14]: 
int                     int64
string                 object
dt        datetime64[ns, UTC]
dtype: object

# we would actually implement like this.
In [15]: pd.Series({l.name: l.dtype for l in mi.levels})                                                                                                                   
Out[15]: 
int                     int64
string                 object
dt        datetime64[ns, UTC]
dtype: object
```

### PR DESCRIPTION
**Pull request by skvrahul**
- [x] closes #37062 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Adds the .dtypes property to MultiIndex using the suggested implementation

### CODE REVIEW

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -191,6 +191,7 @@ Other enhancements
 - :meth:`DatetimeIndex.searchsorted`, :meth:`TimedeltaIndex.searchsorted`, :meth:`PeriodIndex.searchsorted`, and :meth:`Series.searchsorted` with datetimelike dtypes will now try to cast string arguments (listlike and scalar) to the matching datetimelike type (:issue:`36346`)
 - Added methods :meth:`IntegerArray.prod`, :meth:`IntegerArray.min`, and :meth:`IntegerArray.max` (:issue:`33790`)
 - Where possible :meth:`RangeIndex.difference` and :meth:`RangeIndex.symmetric_difference` will return :class:`RangeIndex` instead of :class:`Int64Index` (:issue:`36564`)
+- Added :meth:`MultiIndex.dtypes` (:issue:`37062`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:194]: can you add in the docs in  reference/index.rst as well

File: pandas/tests/indexes/multi/test_get_set.py [hunk 2]
```diff
@@ -27,6 +27,11 @@ def test_get_level_number_integer(idx):
         idx._get_level_number("fourth")
 
 
+def test_get_dtypes(idx):
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_get_set.py:30]: can you add the issue number as a comment

i think idx is too simple, can you construct a MI like in the OP and test it here

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (skvrahul) [reply doc/source/whatsnew/v1.2.0.rst:194]: Do you mean reference/indexing.rst? That's where I see the relevant methods/props documented

**Comment by skvrahul** [comment]
@jreback Made the requested changes. Could you take a look?

**Comment by jreback** [comment]
can you merge master

**Comment by skvrahul** [comment]
> can you merge master

Done.

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (arw2019) [reply doc/source/whatsnew/v1.2.0.rst:223]: yes i think it's `reference/indexing.rst`

**Review by arw2019 — CHANGES_REQUESTED** [review]
@skvrahul can you merge master once more & resolve conflict & address @jreback comment? We'll then re-review

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (skvrahul) [reply doc/source/whatsnew/v1.2.0.rst:223]: I have added `MultiIndex.dtypes` to `reference/indexing.rst` in [this commit](https://github.com/pandas-dev/pandas/pull/37073/commits/de56f049f05e63f661268db914c3241b31ba87bf#)

**Review by arw2019 — COMMENTED** [review]
lgtm

cc @jreback

**Comment by skvrahul** [comment]
I've resolved conflicts. Should be good to merge with the base branch now.

File: pandas/tests/indexes/multi/test_get_set.py [hunk 3]
```diff
@@ -27,6 +29,18 @@ def test_get_level_number_integer(idx):
...
+def test_get_dtypes(idx_multitype):
+    # Test MultiIndex.dtypes (GH-37062)
+    expected = pd.Series(
+        {
+            "int": np.dtype("int64"),
+            "string": np.dtype("O"),
+            "dt": DateTimeTZDtype(tz="utc"),
+        }
+    )
+    assert expected.equals(idx_multitype.dtypes)
```
Comment (arw2019) [inline pandas/tests/indexes/multi/test_get_set.py:41]: use `tm.assert_series_equal`

File: pandas/tests/indexes/multi/test_get_set.py [hunk 4]
```diff
@@ -27,6 +29,18 @@ def test_get_level_number_integer(idx):
         idx._get_level_number("fourth")
 
 
+def test_get_dtypes(idx_multitype):
+    # Test MultiIndex.dtypes (GH-37062)
```
Comment (arw2019) [inline pandas/tests/indexes/multi/test_get_set.py:33]: nitpick: do `# Gh37062`

File: pandas/tests/indexes/multi/conftest.py [hunk 5]
```diff
@@ -24,6 +24,18 @@ def idx():
     return mi
 
 
+@pytest.fixture
+def idx_multitype():
+    # a MultiIndex with several dtypes
+    first_axis = [1, 2, 3]
+    second_axis = list("abc")
+    third_axis = pd.date_range("20200101", periods=2, tz="UTC")
+    mi = MultiIndex.from_product(
```
Comment (arw2019) [inline pandas/tests/indexes/multi/conftest.py:33]: personal taste but I'd stick `first_axis`, `second_axis`, `third_axis` directly into the constructor

File: pandas/core/indexes/multi.py [hunk 6]
```diff
@@ -701,6 +701,15 @@ def array(self):
             "'MultiIndex.to_numpy()' to get a NumPy array of tuples."
         )
 
+    @cache_readonly
+    def dtypes(self):
```
Comment (arw2019) [inline pandas/core/indexes/multi.py:705]: could add a return type annotation

**Review by arw2019 — COMMENTED** [review]
small comments

File: pandas/core/indexes/multi.py (hunk 6 above)
Comment (skvrahul) [reply pandas/core/indexes/multi.py:705]: I'm assuming this is what you mean:
```suggestion
    def dtypes(self) -> Series:
```
However, it looks like we don't necessarily always have access to the **Series** type in this module. Only if the TYPE_CHECKING flag is set.

Comment (arw2019) [reply pandas/core/indexes/multi.py:705]: yeah so I think we can just do `"Series"` with the quotes

File: doc/source/whatsnew/v1.2.0.rst [hunk 7]
```diff
@@ -240,6 +240,7 @@ Other enhancements
 - Calling a NumPy ufunc on a ``DataFrame`` with extension types now preserves the extension types when possible (:issue:`23743`).
 - Calling a binary-input NumPy ufunc on multiple ``DataFrame`` objects now aligns, matching the behavior of binary operations and ufuncs on ``Series`` (:issue:`23743`).
 - Where possible :meth:`RangeIndex.difference` and :meth:`RangeIndex.symmetric_difference` will return :class:`RangeIndex` instead of :class:`Int64Index` (:issue:`36564`)
+- Added :meth:`MultiIndex.dtypes` (:issue:`37062`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:243]: can you move to 1.3 (merge master and its there)

File: pandas/tests/indexes/multi/conftest.py [hunk 8]
```diff
@@ -24,6 +24,16 @@ def idx():
     return mi
 
 
+@pytest.fixture
+def idx_multitype():
```
Comment (jreback) [inline pandas/tests/indexes/multi/conftest.py:28]: no need for this fixture, just create the frame in the test that is using it

File: pandas/tests/indexes/multi/test_get_set.py [hunk 9]
```diff
@@ -1,6 +1,8 @@
 import numpy as np
 import pytest
 
+from pandas.core.dtypes.dtypes import DatetimeTZDtype as DateTimeTZDtype
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_get_set.py:4]: you doing need to alias

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by skvrahul** [comment]
Made the requested changes and passing all tests. Should be okay to merge now
@jreback @arw2019

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @skvrahul very nice!

**Pull request merged**